### DISCUSSION TITLE
Implement HTTPS for Development (localhost)

### DISCUSSION THREAD

**Discussion by stramel** [vercel/next.js#23987] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

I would like to be able to easily use HTTPS on my localhost

## Describe the solution you'd like

Be able to change the `dev` script like so:

```diff
-"dev": "next",
+"dev": "HTTPS=true next",
``` 

## Additional context

CRA and Gatsby both handle this. Gatsby uses `devcert`.

https://www.gatsbyjs.org/docs/local-https/
https://create-react-app.dev/docs/using-https-in-development/

**Comment by frank-dspeed** [comment]
Sounds logical i do the same

**Comment by RayLuxembourg** [comment]
While at it please add example in custom-server/typescript

**Comment by jparklev** [comment]
Any word on this one?

**Comment by frank-dspeed** [comment]
i will do that i issue a pr,

**Comment by sospedra** [comment]
This is very much needed. Most of the modern web API capabilities are restricted to work only under secure connections. Not to mention trying to test in a real device. Accessing `http://192.168.x.y:3000` is subpar compared to `https://{same}`

**Comment by frank-dspeed** [comment]
I closed my effort to implement that into this project it is not worth the work a project member also has closed my PullRequest. I jump into the next framework with another name then next :).

https://github.com/direktspeed/next.js/issues/2 has some useful stuff in it if someone is interested in picking this up.

**Comment by timneutkens** [comment]
> I closed my effort to implement that into this project it is not worth the work a project member also has closed my PullRequest

You only opened an issue title "PR", no pull-request was opened hence why it was closed in favor of this issue which already existed 🤔

**Comment by frank-dspeed** [comment]
@timneutkens its ok for me no need to talk about that vercel is dead for me. I will do a little youtube video about that and then forget it and call it a day.

**Comment by malixsys** [comment]
See [This example](https://github.com/vercel/next.js/blob/canary/examples/with-http2/server.js)...

**Comment by mifrej** [comment]
Is there any plan to follow with this issue?
It would be great to have this feature natively baked in in the nextjs.config instead creating custom Node server like in the mentioned example

**Comment by malixsys** [comment]
I suggest using ngrok in the mean time…

**Comment by valoricDe** [comment]
@timneutkens can you elborate why the feature request got closed? I think an automated process with something like gatsby put in place would be helpful for a wide audience to have such a feature. To have local and prod environment as close as possible. Altough I also understand that one only cares for it much when one is doing performance optimisations (http2)

**Reply by timneutkens** [reply]
It didn't get closed.

<img width="938" alt="Screen Shot 2021-05-05 at 15 26 58" src="https://user-images.githubusercontent.com/6324199/117148269-5ebc8b80-adb6-11eb-9422-c332505a029e.png">

**Reply by valoricDe** [reply]
Ok, seems like I am not familiar with the new github instuments. I thought only issues and pr's are actionable items. Thx for clarification

**Comment by malixsys** [comment]
**WORKAROUND**: Just add a script to your `package.json`:
```json
    "ngrok": "npx ngrok http -bind-tls=true localhost:3000",
```
🤷🏽‍♂️

**Reply by valoricDe** [reply]
Thank you! That also doesn't interfere with vercel 🙏

**Comment by frank-dspeed** [comment]
## If your using chrome
you can turn http into https without any additional stuff in next.js
go to
```
chrome://flags/#allow-insecure-localhost
```

and enable that it will handle your http://localhost as if it would be https.

**Comment by jardicc** [comment]
How do I change the configuration for `next dev` to use `server.js` where I do specify my own certificates?
